modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C

```diff
@@ -928,5 +944,136 @@ PolygonConcentricCircleMeshGeneratorBase::generate()
   if (flat_side_up)
     MeshTools::Modification::rotate(*mesh0, 180.0 / (Real)_num_sides, 0.0, 0.0);
   mesh0->set_isnt_prepared();
+
+  if (_has_rings && getParam<bool>("replace_inner_ring_with_delaunay_mesh"))
+  {
+    if (getParam<bool>("quad_center_elements"))
+      paramWarning("quad_center_elements",
+                   "ignored because the inner ring will be replaced by a Delaunay mesh with "
+                   "'replace_inner_ring_with_delaunay_mesh=true'");
+
+    // remove elements of the inner ring and create an inner-ring external boundary side set
+    std::set<Elem *> deleteable_elems;
+    for (auto & elem : mesh0->element_ptr_range())
+      if (elem->vertex_average().norm() < ring_radii_corr[0])
+        deleteable_elems.insert(elem);
+
+    const boundary_id_type boundary_id = MooseMeshUtils::getBoundaryIDs(*mesh0, {"_foo"}, true)[0];
+    BoundaryInfo & boundary_info = mesh0->get_boundary_info();
+    for (auto & elem : deleteable_elems)
+    {
+      unsigned int n_sides = elem->n_sides();
+      for (unsigned int n = 0; n != n_sides; ++n)
+      {
+        Elem * neighbor = elem->neighbor_ptr(n);
+        if (!neighbor)
+          continue;
+
+        const unsigned int return_side = neighbor->which_neighbor_am_i(elem);
+
+        if (neighbor->neighbor_ptr(return_side) == elem)
+        {
+          neighbor->set_neighbor(return_side, nullptr);
+          boundary_info.add_side(neighbor, return_side, boundary_id);
+        }
+      }
+
+      mesh0->delete_elem(elem);
+    }
+
+    // build the 1d mesh from the new boundary
```

Comment:
the next 40 lines look like they should be a utility instead of being coded here

Reply:
If we are to make this a utility, these lines will need to be coordinated with the existing code in `LowerDBlockFromSidesetGenerator` to be a single utility.

Same thing could be done the for elements deletion lines above along with the corresponding code in `ElementDeletionGeneratorBase`